### ISSUE TITLE
Mono support

### ISSUE THREAD

**Issue by adamralph** [config-r/config-r#186]

**Comment by christianhuening** [comment]
Hi there!
Is there an update regarding mono support? Since we are currently trying to use ConfigR with Mono through DNX and are running into problems when it comes to loading settings from our configuration file. The same code works, wenn being used through dnx and Visual Studio 2015 on Windows.

cheers,
Christian

**Comment by adamralph** [comment]
Hi @christianhuening, I tried to get this done on my flight back from .NET Fringe but I hit a [dead end](https://github.com/adamralph/config-r/commit/991a0f8cfaad986d0d0839bd67d69743dcb2a081). Unfortunately I can no longer remember what the exact problem was and I didn't put anything more into the commit message but I do remember thinking it was a show stopper.

I also remember thinking that it will be better to just wait until scriptcs switches from the old CTP to a newer, Mono compatible version of Roslyn but I'm not sure that's going to happen any time soon. Work is under way to [switch scriptcs to Roslyn 1.0.0 RC2](https://github.com/scriptcs/scriptcs/pull/1064) but AFAIK that does not run under Mono either.

So, in short, I can't currently figure out a way to get this done. All help welcome :wink:.

**Comment by devlead** [comment]
Could you use Mono scripting like we did in cake-build/cake#280

**Comment by adamralph** [comment]
@devlead the scriptcs Mono engine does exactly that, but I hit a dead end trying to make ConfigR use it.

I guess I should really revisit the branch and find out exactly what the problem was and provide details here. That might in a couple of weeks though. I'm going away for 2 weeks tomorrow morning and I don't think I'll get time tonight.

**Comment by devlead** [comment]
Ok cool:+1:

**Comment by filipw** [comment]
currently scriptcs mono support doesn't allow custom hosts unless the host is emitted at runtime using IL (which is doable but crazy).

anyway, IMHO this can be superseded by #222

**Comment by adamralph** [comment]
Yeah, I'm expecting #222 to trump this...

**Comment by adamralph** [comment]
Putting this on hold in light of #222. Probably will end up being withdrawn.

**Comment by adamralph** [comment]
This is all set to change with .NET Core so I don't think it's worth investing any time in this. I also think that, ultimately, support for all platforms will be provided via support for the relevant [.NET Standard](https://gist.github.com/davidfowl/8939f305567e1755412d6dc0b8baf1b7), so I don't think explicit support for anything other than net45/net46 will ever be a thing in ConfigR. For that reason, I'm going to close this issue.
